src/http_server.cc

```diff
@@ -3055,11 +3055,30 @@ HTTPAPIServer::StartTrace(
     TRITONSERVER_InferenceTrace** triton_trace)
 {
 #ifdef TRITON_ENABLE_TRACING
+  TraceStartOptions start_options;
+  InferenceTraceMode mode = TRACE_MODE_TRITON;
+  trace_manager_->GetTraceMode(model_name, &mode);
+  if (mode == TRACE_MODE_OPENTELEMETRY) {
+#ifndef _WIN32
+    const HttpTextMapCarrier carrier(req->headers_in);
+    auto prop =
+        otel_cntxt::propagation::GlobalTextMapPropagator::GetGlobalPropagator();
+    auto ctxt = opentelemetry::context::Context();
+    start_options.propagated_context = prop->Extract(carrier, ctxt);
+    otel_trace_api::SpanContext span_context =
+        otel_trace_api::GetSpan(start_options.propagated_context)->GetContext();
+    if (span_context.IsValid()) {
```

Comment:
Will this always be evaluated to `true` in the case of no client context? And causes force sampling as long as the trace is in OpenTelemetry mode?

Reply:
I've touched this in the above comment. When no context is propagated, and invalid span will be stored in `start_options.propagated_context`, so `span_context.IsValid()` will return false.

This behavior is tested in `test_http_trace_triggered`, where I set `rate=5` and send 5 requests without `otel` headers -> only one is traced, then send 5 requests with `otel` headers and check that all 5 are traced